src/pages/ActivityPage/ActivityPage.tsx

```diff
@@ -132,29 +133,42 @@ const ActivityPage = () => {
             />
           </div>
         ) : (
-          <RoadmapList
-            userId={id}
-            onDeleteRoadmap={handleDeleteRoadmap}
-            roadmapList={roadmapList}
-          />
-        )}
-        {!isLoading && !!roadmapList.length ? (
-          <Pagination
-            onPageChange={handlePageChange}
-            totalPage={totalPage}
-            currentPage={currentPage}
-            containerClassName={style.pagination}
-          />
-        ) : (
-          <p className={style.info}>
-            <Link
-              to={ROUTES.ROADMAPS}
-              className={buttonVariant({ variant: 'link' })}
-            >
-              ADD
-            </Link>
-            roadmap to start track your progress
-          </p>
+          <>
+            {userActivity.added === 0 ? (
+              <p className={style.info}>
+                <Link
+                  to={ROUTES.ROADMAPS}
+                  className={buttonVariant({ variant: 'link' })}
+                >
+                  ADD
+                </Link>
+                roadmap to start track your progress
+              </p>
+            ) : roadmapList.length ? (
+              <>
+                <RoadmapList
+                  userId={id}
+                  onDeleteRoadmap={handleDeleteRoadmap}
+                  roadmapList={roadmapList}
+                />
+                {totalPage > 1 && (
+                  <Pagination
+                    onPageChange={handlePageChange}
+                    totalPage={totalPage}
+                    currentPage={currentPage}
+                    containerClassName={style.pagination}
+                  />
+                )}
+              </>
+            ) : (
+              <div className={style.notFound}>
+                <Icon id="icon-material" size={28} />
+                <p className={style.notFoundText}>
+                  Nothing was found for your filters
+                </p>
+              </div>
+            )}
+          </>
```

Comment:
A little difficult read several ternary operator in one expression. Please split them to logical parts